### PR TITLE
Devise_authentication

### PR DESCRIPTION
**Pull request by erezfree29**
created a members-only app using the devise gem which includes 

-User sign up with a secure password
-User sign in
-User sign-up validations(user creation)
-Post creation only for signed-up users.
-Post validations on creating.
-Restricted access to post creation only to logged-in users.
-Access to post-index where all the posts are shown.
-Posts Authors are only displayed if a user is signed in.

### CODE REVIEW

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: 
-------------------
Hello @erezfree29  You did an amazing job so far :+1: :clap: 

TO HIGHLIGHT: 
- Your app is working properly. 
- Your repository is professional.
- Your Project is professional
- I like that the password must be difficult so is difficult to know the password. 

However, there are a few changes to make to complete this project :hammer_and_wrench: see them below :eyes: :point_down: 
--------------------
- [ ] Please check the step on how to clone the repo. You write `cd re-form` and it must be `cd Authentication`
- [ ] The issue link is not working. 
- [ ] Please add steps on your readme on how to start the project. E.g `rails server or rails s` and `http://localhost:3000/` 
- [ ] Please check the title of your pull request you misspell authentication. 
- [ ] Please don't use logic in your views instead put the logic on helpers. [Here](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) you can read about how to put your logic on helpers. 

no_logic_in_helpers_requirement

- [ ] Please add a sign-in link on post views. 

sign_in_out_link_requirement

- [ ] On your routes, please scope only the routes you need. E.g `resources :posts, only: [:new, :create, :index]`.
- [ ] Please write the sign in and sign out in `application.html.erb` that is inside the `layouts` folder so you don't need to manually repeat it in every view. 
------------
Make these required changes and send it again for another review. 
Reachme on SLACK if you have any questions about the requirements. 
@ Brian Sammit Cruz Rodriguez
HAPPY CODING!!

**Comment by mimipeshy** [comment]
## STATUS: CHANGES REQUESTED 🔴 ♻️ 
Hello, @erezfree29 👋🏾 this is a great job you have done in this milestone. You have worked on the TSE comments which is commendable. However, before approval, let's fix the following ⬇️ 

- [ ]  Kindly remove logic from your views. Logic includes loops and if statements. This will ensure your project follows the MVC pattern. You can create all your logic inside the helpers.  Read about [helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)


That's all 😄, kindly submit another review when done
If you need any clarification, kindly post your comment here and tag me @mimipeshy and will be glad to help

Regards,
Peris Ndanu

File: app/views/posts/index.html.erb
Comment (mimipeshy) [inline app/views/posts/index.html.erb]: Ensure there is absolutely no logic in your views

**Review by oloomoses — APPROVED** [review]
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Great job working on the suggested changes from the previous review.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

**Pull request merged**
